core/src/main/java/org/testcontainers/containers/wait/strategy/HttpWaitStrategy.java

```diff
@@ -45,7 +52,17 @@
      * @return this
      */
     public HttpWaitStrategy forStatusCode(int statusCode) {
-        this.statusCode = statusCode;
+        statusCodes.add(statusCode);
+        return this;
+    }
+
+    /**
+     * Waits for the response to pass the given predicate
+     * @param statusCodePredicate The predicate to test the response against
+     * @return this
+     */
+    public HttpWaitStrategy forStatusCodeMatching(Predicate<Integer> statusCodePredicate) {
```

Comment:
I'm thinking about the use case where somebody will call `Wait.forHttp().forStatusCode(200).forStatusCodeMatching(it -> it > 400)`. Maybe we should throw an error, or handle both statuses?

Reply:
I do really like the Predicate approach here - it opens the door to having some useful ready-to-use predicates in the future (e.g. `anySuccessfulStatusCode()` or `anyServerErrorStatusCode()` etc...

Reply:
With the latest change below I'm now hopefully supporting both.